ldap/servers/slapd/opshared.c

```diff
@@ -892,7 +896,7 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
                 slapi_pblock_get(pb, SLAPI_RESULT_CODE, &err);
                 /* PAGED RESULTS */
                 /* Explicit ctrlp test avoid gcc -fanalyzer warning */
-                if (op_is_pagedresults(operation)) {
+                if (op_is_pagedresults(operation) && !flag_referral) {
```

Comment:
I have a doubt here. My understanding is that calling be_search_results_release/delete_search_result_set with a NULL result set lead to sigsev. Am I correct ?
Would it be possible to harden delete_search_result_set so that it does not crash ?